lib/i18n/backend/base.rb

```diff
@@ -150,6 +150,8 @@ def pluralize(locale, entry, count)
         def interpolate(locale, string, values = {})
           if string.is_a?(::String) && !values.empty?
             I18n.interpolate(string, values)
+          elsif string.is_a?(::Array) && !values.empty?
+            string.map { |el| interpolate(locale, el, values) }
           else
             string
           end
```

Comment:
IMO this whole method should be rewritten to make it clearer and cut down repetition:

``` ruby
def interpolate(locale, subject, values = {})
  return subject if values.empty?

  case subject
  when ::String then I18n.interpolate(subject, values)
  when ::Array then subject.map { |element| interpolate(locale, element, values) }
  else
    subject
  end
end
```